src/gluonts/trainer/_base.py

```diff
@@ -311,43 +316,60 @@ def loop(
                         logger.info("Stopping training")
                         break
 
-                    if loss_value(epoch_loss) < best_epoch_info.metric_value:
-                        best_epoch_info = BestEpochInfo(
-                            params_path="%s-%04d.params"
-                            % (base_path(), epoch_no),
-                            epoch_no=epoch_no,
-                            metric_value=loss_value(epoch_loss),
-                        )
+                    if loss_value(epoch_loss) < best_epoch_info["score"]:
+                        bp = base_path()
+                        best_epoch_info = {
+                            "params_path": f"{bp}-0000.params",
+                            "epoch_no": epoch_no,
+                            "score": loss_value(epoch_loss),
+                        }
+
                         net.save_parameters(
-                            best_epoch_info.params_path
+                            best_epoch_info["params_path"]
```

Comment:
Perhaps we can simplify the trainer logic a bit:

1. Checkpoint after every epoch no matter if we do model averaging or not (should not be a big overhead)
2. Use an `AveragingStrategy` similar to what you already introduced below. If we don't want model averaging this just picks the best epoch, or the last epoch etc

I'm also fine with merging as is for now. We can do these changes later.